src/components/explore/AggregationsOverTimeGraph.svelte

```diff
@@ -143,6 +135,64 @@
     // Finally, set the flag to open the context menu.
     $showContextMenu = true; // eslint-disable-line no-unused-vars
   }
+
+  function getDefaultReferencePoint() {
+    if ($store.ref) {
+      const found = data.find((d) => d[aggregationLevel] === $store.ref);
+      if (found) return found;
+    }
+    return data[data.length - 1];
+  }
+
```

Comment:
There's a lot happening in these 3 functions below. Adding a generalized comment above each one, and some comments within them to explain what is going on would be really helpful for future maintainers and ourselves.